src/game/stage/player.c

```diff
@@ -794,9 +794,9 @@ void InitializePlayer(Player *p)
     *u99_r6 = 0x7F;
 #else
     {
+        u32 *ptr = (u32 *)(&p->SA2_LABEL(unk99)[0]);
         playerID = &p->playerID;
```

Comment:
Also, if you really think this loop is real code, you might as well just wrap the `if ((*playerID == 0)` in non matches too and not use the pointers if you are not using the asm. The code you have here is like half way between a fake a real match.